pkg/apis/experimental/types.go

```diff
@@ -508,35 +511,70 @@ type IngressStatus struct {
 	LoadBalancer api.LoadBalancerStatus `json:"loadBalancer,omitempty"`
 }
 
-// IngressRule represents the rules mapping the paths under a specified host to the related backend services.
-type IngressRule struct {
-	// Host is the fully qualified domain name of a network host, or its IP
-	// address as a set of four decimal digit groups separated by ".".
-	// Conforms to RFC 1738.
+// IngressHost represents the rules mapping the paths under a specified host to
+// the related backend services.
+type IngressHost struct {
+	// Host is the fully qualified domain name of a network host, as defined
```

Comment:
hmm, technically yes but that is a valid hostname, but no TLD is purely
numeric.  If we're asserting that this is a hostname field and someone
enters an IP, we 're going to do a DNS lookup on it.

On Fri, Sep 25, 2015 at 5:44 PM, Prashanth B <notifications@github.com>
wrote:

> In pkg/apis/experimental/types.go
> <https://github.com/kubernetes/kubernetes/pull/14459#discussion_r40487631>
> :
>
> > @@ -508,35 +511,70 @@ type IngressStatus struct {
> >  	LoadBalancer api.LoadBalancerStatus `json:"loadBalancer,omitempty"`
> >  }
> >
> > -// IngressRule represents the rules mapping the paths under a specified host to the related backend services.
> > -type IngressRule struct {
> > -	// Host is the fully qualified domain name of a network host, or its IP
> > -	// address as a set of four decimal digit groups separated by ".".
> > -	// Conforms to RFC 1738.
> > +// IngressHost represents the rules mapping the paths under a specified host to
> > +// the related backend services.
> > +type IngressHost struct {
> > +	// Host is the fully qualified domain name of a network host, as defined
>
> wait, a string with ip is also a subdomain right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14459/files#r40487631>.
>